delocate/tools.py

```diff
@@ -812,22 +813,42 @@ def add_rpath(filename: str, newpath: str, ad_hoc_sign: bool = True) -> None:
         replace_signature(filename, "-")
 
 
-def zip2dir(zip_fname: str, out_dir: str) -> None:
+def zip2dir(
+    zip_fname: str | PathLike[str], out_dir: str | PathLike[str]
+) -> None:
     """Extract `zip_fname` into output directory `out_dir`
 
     Parameters
     ----------
-    zip_fname : str
+    zip_fname : str or Path-like
         Filename of zip archive to write
-    out_dir : str
+    out_dir : str or Path-like
         Directory path containing files to go in the zip archive
     """
-    # Use unzip command rather than zipfile module to preserve permissions
+    # The zipfile module does not preserve permissions correctly
     # http://bugs.python.org/issue15795
-    _run(["unzip", "-q", "-o", "-d", out_dir, zip_fname], check=True)
-
-
-def dir2zip(in_dir, zip_fname):
+    with zipfile.ZipFile(zip_fname, "r") as zip:
```

Comment:
Is this code from auditwheel?

Reply:
`zip2dir` and `dir2zip` were taken from auditwheel, [specifically from here](https://github.com/pypa/auditwheel/blob/321292d620f14ddc7ad3247d71746679d6e51558/src/auditwheel/tools.py#L31-L91), and then modified.  Delocate has different requirements from permission bits and timestamps.

The docstrings and parameters are very similar which makes it look like it's been copied before, in one direction or the other or from the same sources.